src/OCRNetEngine.cpp

```diff
@@ -177,6 +202,43 @@ OCRNetEngine::infer(BufferManager& buffer_mgr, std::vector<std::pair<std::string
             temp_de_texts.emplace_back(std::make_pair(de_text, prob));
         }
     }
+    else if (mDecodeMode == CLIP)
```

Comment:
I don't get this decode logic --- why do you need to transform `tmp_text` to `de_text` ?

Reply:
the tmp_text is the raw model output which include 94 chars from training set `"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~",`  in decode, we need filter the raw output text and only keep the char which is in test char set `"0123456789abcdefghijklmnopqrstuvwxyz"`, that's why I save the raw output to `tmp_text` and then filter it to the final `de_text`

Reply:
I don't think this is necessary. We can just keep the raw output from `OCRNetEngine`, if the user need filter on the output, they can add it after they get the output from nvOCDR.

Reply:
make scene, how about to make this filter as a option and create a config for this filter

Reply:
I would prefer you implement a sample filter in the c++ sample. Keep the core lib simple and clean.